### ISSUE TITLE
Release Twisted 15.0

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7335]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7335|
|Type|task|
|Created|2014-05-12 11:23:49Z|
|Branch|https://github.com/twisted/twisted/tree/releases/release-15.0.0-7335|

Onwards to -14.1- 15.0! 

<details><summary>Searchable metadata</summary>

```
trac-id__7335 7335
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_releases_release_15_0_0_7335 branches/releases/release-15.0.0-7335
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1399893829000000 1399893829000000
changetime__1445217216921098 1445217216921098
version__None None
owner__hawkowl hawkowl
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

*does celebration trombones* review please!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

I took a look at diff and it looks good. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set owner to @hawkowl|
|-|-|

The release notes for 14.0.1 and 14.0.2 should really be merged in here, since from a user's perspective there is a linear timeline of features here.  I guess that means this will need one more cycle of review.  Sorry :-\.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_9"></a>|@hawkowl removed owner|
|-|-|

Merged in the changelogs.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

> The release notes for 14.0.1 and 14.0.2 should really be merged in here

Is that a documented part of the release process?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

Replying to [exarkun](#note_10):
> > The release notes for 14.0.1 and 14.0.2 should really be merged in here
> 
> Is that a documented part of the release process?

I'm not sure. They *should* have been there already (since releases "should" be merged into trunk), but the original idea was that 14.0.2 was a hotfix, and 14.1/15.0 would include an entirely different fix than 14.0.2 did. But... it didn't, and in merging the changelogs I opted to have 14.0.1/2 as what was released, and everything else was in 15.0.

How should we handle this possible divergence next time? What's a fix, and what is fixed in trunk, may be different, and so a x.y.1 may be an entirely different "line" of versions than x.y+1.0. But, version numbers are supposed to exist on a concurrent plane... I'm not sure.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

> But... it didn't, and in merging the changelogs I opted to have 14.0.1/2 as what was released, and everything else was in 15.0.

I don't understand this.  You merged 14.0.1 and 14.0.2 into trunk?  Into the 15.0 release branch?  And resolved conflicts in the news files?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl commented|
|-|-|

Replying to [exarkun](#note_12):
> > But... it didn't, and in merging the changelogs I opted to have 14.0.1/2 as what was released, and everything else was in 15.0.
> 
> I don't understand this.  You merged 14.0.1 and 14.0.2 into trunk?  Into the 15.0 release branch?  And resolved conflicts in the news files?

I copied in the 14.0.1/14.0.2 news file entries (manually) into the release branch's notes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

So by "should have been there" you didn't mean you expected them to be there.  Instead, you meant that the normal release process would have caused them to be there but the 14.0.1 and 14.0.2 releases didn't follow the normal release process?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_15"></a>|@hawkowl commented|
|-|-|

Replying to [exarkun](#note_14):
> So by "should have been there" you didn't mean you expected them to be there.  Instead, you meant that the normal release process would have caused them to be there but the 14.0.1 and 14.0.2 releases didn't follow the normal release process?

Yeah, 14.0.1 and 14.0.2 didn't follow it -- there weren't prereleases or reviews prior to getting a fix deployed, and it didn't get merged in as the problem that was fixed was put in a ticket to make it into trunk. I think what happens in a security fix situation (of which this seemed to be the first one, or at least the only one since 2008, looking at https://twistedmatrix.com/Releases/) needs to be written down, so what to do in this situation has a clear answer, and isn't something we're sorting out in the review stage.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

Can you write down what _was_ done and the associated problems?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_17"></a>|@hawkowl commented|
|-|-|

Sure. Is this a release blocker, or something I should do after 15.0 is out the door?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_18"></a>|@glyph commented|
|-|-|

Replying to [hawkowl](#note_17):
> Sure. Is this a release blocker, or something I should do after 15.0 is out the door?

It's not a blocker - release blockers are for things that somehow threaten the quality of the release (regressions, pretty much), and the output here is a small amount of prose in some text files we can easily manually verify.

This isn't even a case of parts of the release process being undocumented and secret and thereby broken: 14.0.1 and 2 were anomalous releases, done under a sort of pilot release program for security updates, and maybe not even done in the best way.  The worst case scenario of not documenting this is that we will have to do all the work to figure out a viable security release process again from scratch, and maybe we have to do that anyway.

Nevertheless, the postmortem should totally be done, though, because we _should_ have a totally seamless process for that in the future which should not necessitate manual editing.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_19"></a>|@glyph set owner to @hawkowl|
|-|-|

And, speaking of manually reviewing that prose, i've now done so.  I did some experiments building locally, I see a happy buildbot, I think we are good to go for fifteen dot oh.

```
 ____  _   _ ___ ____    ___ _____ 
/ ___|| | | |_ _|  _ \  |_ _|_   _|
\___ \| |_| || || |_) |  | |  | |  
 ___) |  _  || ||  __/   | |  | |  
|____/|_| |_|___|_|     |___| |_|  
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_20"></a>|@hawkowl set status to `closed`|
|-|-|

(In [43795]) Merge releases/release-15.0.0-7335: Release Twisted 15.0.0

Author: hawkowl
Reviewer: glyph
Fixes: [#7335](https://github.com/twisted/twisted/issues/7335)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_21"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone deleted
